### PR TITLE
Distribute on PyPI

### PR DESCRIPTION
**Pull request by cle1109**
This PR addresses #46. I updated `setup.py`, but I have some questions. For starters, which email address should we use? Ideally, I would like to have an email address where (selected) members of the SCoT development team have access. I don't think GitHub support this?

Any other comments on this file (I used [this](https://github.com/pypa/sampleproject/blob/master/setup.py) as a template).

### CODE REVIEW

**Comment by kazemakase** [comment]
I don't think GitHub provides email addresses.

Maybe we can create a gmail (or something like that) address and put the login data somewhere on github where only selected members can see it?

**Comment by cle1109** [comment]
OK. What about creating a mailing list at Google Groups and forward mail to this address to project members? Something like scot-dev@googlegroups.com?

**Comment by kazemakase** [comment]
I like the mailing list idea better than a shared account, but I guess it still requires account management.

**Comment by cle1109** [comment]
At least that's how they do it in the [sample project](https://github.com/pypa/sampleproject/blob/master/setup.py).

**Comment by cle1109** [comment]
OK, I created a collaborative inbox (aka mailling list) on Google Groups. Our email address is scotdev@googlegroups.com. OK?

**Comment by cle1109** [comment]
I uploaded it to [TestPyPI](https://testpypi.python.org/pypi) - can you please check it (as long as it's there)? @agramfort, maybe you can also comment if the package (distribution) is OK like that.

**Comment by kazemakase** [comment]
We should change the description to something along the line of "EEG source connectivity toolbox", because just source connectivity is misleading in a programming context.

File: setup.py [hunk 1]
```diff
@@ -1,17 +1,38 @@
...
+        'Programming Language :: Python :: 2.6',
+        'Programming Language :: Python :: 2.7',
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.2',
+        'Programming Language :: Python :: 3.3',
+        'Programming Language :: Python :: 3.4',
+    ],
+    keywords='source connectivity EEG MEG ICA',
+    packages=['scot', 'scot.eegtopo', 'scot.external'],
+    install_requires=['numpy >=1.7', 'scipy >=0.12'],
```
Comment (agramfort) [inline setup.py]: don't add these deps. It may lead to pip installing numpy and scipy and you don't want this. It's likely to break the install of people.

Comment (cle1109) [inline setup.py]: Why don't we want pip installing numpy and scipy? I thought that's exactly what we want, because otherwise scot does not work.

Comment (agramfort) [inline setup.py]: import numpy will fail an people are free to install it correctly : apt-get
or conda

pip wheels exist but It's not yet mature to me.

Comment (cle1109) [inline setup.py]: OK. On my Mac, I always used pip to install numpy and scipy and never had any problems. I don't know about Linux though, but you are right, on Linux I would also prefer installing these packages through the distribution's package manager and not via pip.

Comment (kazemakase) [inline setup.py]: +1 for removing the requirement.

You never know against which blas/lapack libraries pip trys to link numpy (maybe even some included ones). I had big problems with that when trying to distribute my computations over a number of different debian/ubuntu systems in the lab.

**Comment by cle1109** [comment]
I think it would be nice to finally upload scot to PyPI. I removed the dependencies and uploaded the updated version to [TestPyPI](https://testpypi.python.org/pypi/scot/0.1.1). It looks a bit messy, so I guess we need to clean up some files so that the description renders nicely. Also, I had to increase the version number, because TestPyPI wouldn't let me upload an updated version with the same version number.

Any comments?

**Comment by kazemakase** [comment]
I don't litke the style of PyPi anyway, so I guess in that light it looks okay :)

The short description (near the top) appears twice:
> EEG/MEG Source Connectivity Toolbox
EEG/MEG Source Connectivity Toolbox for Python.

Also, according to GRMP, the toolbox is not *for* Python. Instead it is *for* doing connectivity, and written *in* Python...

**Comment by kazemakase** [comment]
I don't like the PyPi style anyway. Guess it's fine the way it is now...

Two comments:
1. Why does the short discription appear twice (near the top)?
> EEG/MEG Source Connectivity Toolbox
EEG/MEG Source Connectivity Toolbox for Python.

2. According to GRMP, the toolbox is not *for* Python. Instead, it is *for* doing connectivity while being written *in* Python... I'm not sure if I agree, though :)

**Comment by cle1109** [comment]
OK - maybe we can copy elements from [scikit-learn](https://pypi.python.org/pypi/scikit-learn/0.16.1)? Their PyPI site looks nice.

Regarding for/in, I don't really care. Let's just drop it and say that SCoT is a Python module for source connectivity estimation. On that note, is it a module or a package :-)?

**Comment by kazemakase** [comment]
[It's a package](http://stackoverflow.com/a/7948504)

**Comment by kazemakase** [comment]
Btw, why does this PR have merge conflicts?

File: setup.py [hunk 2]
```diff
@@ -1,17 +1,37 @@
...
+    author='SCoT Development Team',
+    author_email='scotdev@googlegroups.com',
+    license='MIT',
+    classifiers=[
+        'Development Status :: 4 - Beta',
+        'Intended Audience :: Science/Research',
+        'Topic :: Scientific/Engineering',
+        'License :: OSI Approved :: MIT License',
+        'Programming Language :: Python :: 2',
+        'Programming Language :: Python :: 2.6',
```
Comment (kazemakase) [inline setup.py]: I don't think scot works with Python 2.6.

File: setup.py [hunk 3]
```diff
@@ -1,17 +1,37 @@
...
+        'Programming Language :: Python :: 2.7',
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.2',
+        'Programming Language :: Python :: 3.3',
+        'Programming Language :: Python :: 3.4',
```
Comment (kazemakase) [inline setup.py]: Although I'm sure scot works with Python 3.4, we only test on 3.3.
Should we adjust this here?
Should we update our tests?

Comment (cle1109) [inline setup.py]: We should support the latest versions of 2 and 3, which are 2.7 and 3.4 at the moment. So yes, let's update our tests.

File: setup.py (hunk 2 above)
Comment (cle1109) [inline setup.py]: OK, we can drop it as long as we keep support for 2.7.

**Comment by cle1109** [comment]
There are conflicts in README.md.

**Comment by cle1109** [comment]
What do you think? https://testpypi.python.org/pypi/scot/0.1.3

**Comment by kazemakase** [comment]
LGTM

**Comment by cle1109** [comment]
Yes, I've already summarized the steps in a local doc. Since this is only relevant for the core maintainers (us two :-)), where should we put it?

**Comment by kazemakase** [comment]
Let's put it under distributions/pypi/.
Is it possible to create a script for automating the process?

**Comment by cle1109** [comment]
Yes, definitely. I will add this script right now.

**Comment by cle1109** [comment]
Is the README sufficient, or do we need a script? It's more or less just 2 lines. Can you try if it works for you (maybe upload it to TestPyPI under a different name)?

**Pull request merged**